test/interpreter/function-covariance.p.spec.ts

```diff
@@ -0,0 +1,81 @@
+import {HyperFormula} from '../../src'
+import {ErrorType} from '../../src/Cell'
+import {ErrorMessage} from '../../src/error-message'
+import {adr, detailedError} from '../testUtils'
+
+describe('COVARIANCE.P', () => {
+  it('validates number of arguments',  () => {
+    const engine =  HyperFormula.buildFromArray([
+      ['=COVARIANCE.P(B1:B5)'],
+      ['=COVARIANCE.P(B1:B5, C1:C5, D1:D5)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+    expect(engine.getCellValue(adr('A2'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+  })
+
+  it('ranges need to have same amount of elements',  () => {
+    const engine =  HyperFormula.buildFromArray([
+      ['=COVARIANCE.P(B1:B5, C1:C6)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedError(ErrorType.NA, ErrorMessage.EqualLength))
```

Comment:
For empty array there will be different error? "If either array1 or array2 is empty, COVARIANCE.P returns the #DIV/0! error value."

https://support.microsoft.com/en-us/office/covariance-p-function-6f0e1e6d-956d-4e4b-9943-cfef0bf9edfc

Reply:
fixed